parsl/executors/globus_compute.py

```diff
@@ -0,0 +1,166 @@
+
+    """
+
+    def __init__(
+            self,
+            endpoint_id: Optional[UUID_LIKE_T] = None,
+            task_group_id: Optional[UUID_LIKE_T] = None,
+            resource_specification: Optional[Dict[str, Any]] = None,
+            user_endpoint_config: Optional[Dict[str, Any]] = None,
+            label: str = "GlobusComputeExecutor",
+            batch_size: int = 128,
+            amqp_port: Optional[int] = None,
+            client: Optional[Client] = None,
+            **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+
+        endpoint_id:
+            id of the endpoint to which to submit tasks
+
+        task_group_id:
+            The Task Group to which to associate tasks.  If not set,
+            one will be instantiated.
+
+        resource_specification:
+            Specify resource requirements for individual task execution.
+
+        user_endpoint_config:
+            User endpoint configuration values as described
+            and allowed by endpoint administrators. Must be a JSON-serializable dict
+            or None. Refer docs from `globus-compute
+            <https://globus-compute.readthedocs.io/en/latest/endpoints/endpoints.html#templating-endpoint-configuration>`_
+            for more info.
+
+        label:
+            a label to name the executor
+
+        batch_size:
+            the maximum number of tasks to coalesce before
+            sending upstream [min: 1, default: 128]
+
+        amqp_port:
+            Port to use when connecting to results queue. Note that the
+            Compute web services only support 5671, 5672, and 443.
+
+        client:
+            instance of globus_compute_sdk.Client to be used by the executor.
+            If not provided, the executor will instantiate one with default arguments.
+
+        kwargs:
+            Other kwargs listed will be passed through to globus_compute_sdk.Executor
+            as is
+        """
+        super().__init__()
+        self.endpoint_id = endpoint_id
+        self.task_group_id = task_group_id
+        self.resource_specification = resource_specification
+        self.user_endpoint_config = user_endpoint_config
+        self.label = label
+        self.batch_size = batch_size
+        self.amqp_port = amqp_port
+        self.client = client
+
+        if not _globus_compute_enabled:
+            raise OptionalModuleMissing(
+                ['globus-compute-sdk'],
+                "GlobusComputeExecutor requires globus-compute-sdk installed"
+            )
+
+        self._executor: Executor = Executor(
+            endpoint_id=endpoint_id,
+            task_group_id=task_group_id,
+            resource_specification=resource_specification,
+            user_endpoint_config=user_endpoint_config,
+            label=label,
+            batch_size=batch_size,
+            amqp_port=amqp_port,
+            client=self.client,
+            **kwargs
+        )
+
+    def start(self) -> None:
+        pass
```

Comment:
In the Parsl configuration model, this separation of init and start means that the code will try to connect to an executor at the point that a user defines a configuration that uses globus compute, rather than at the point that they initialize Parsl with a Globus compute.  I'm not clear on what this Executor does at the point that it is instantiated (for example, making network connections?).

That might be out of line with other code which doesn't do anything until it is `start`ed - for example, when is the interchange started in a HighThroughputExecutor configuration?

Reply:
I've moved the actual `globus_compute_sdk.Executor` instantiation to start.